library/Icingadb/Web/Controller.php

```diff
@@ -349,19 +360,82 @@ public function createViewModeSwitcher()
             $viewModeSwitcher->setDefaultViewMode($viewMode);
         }
 
-        $viewModeSwitcher->populate(['view' => $this->params->shift($viewModeSwitcher->getViewModeParam())]);
+        $viewModeSwitcher->populate([
+            'view' => $this->params->shift($viewModeSwitcher->getViewModeParam())
+        ]);
 
-        $viewModeSwitcher->on(ViewModeSwitcher::ON_SUCCESS, function (ViewModeSwitcher $viewModeSwitcher) use ($prefs) {
-            $viewMode = $viewModeSwitcher->getValue($viewModeSwitcher->getViewModeParam());
+        if (
+            $viewModeSwitcher->getViewMode() === 'minimal'
+            || Url::fromRequest()->getParam(ViewModeSwitcher::DEFAULT_VIEW_MODE_PARAM) === 'minimal'
+        ) {
+            $hasLimitParam = Url::fromRequest()->hasParam($limitControl->getLimitParam());
 
-            $icingadbPrefs = $prefs->icingadb ?: [];
-            $icingadbPrefs['view_mode'] = $viewMode;
-            $prefs->icingadb = $icingadbPrefs;
+            if ($limitControl->getDefaultLimit() <= LimitControl::DEFAULT_LIMIT && ! $hasLimitParam) {
+                $limitControl->setDefaultLimit($limitControl->getDefaultLimit() * 2);
+            }
 
-            $this->redirectNow(Url::fromRequest()->setParam($viewModeSwitcher->getViewModeParam(), $viewMode));
-        });
+            if ($paginationControl->getDefaultPageSize() <= LimitControl::DEFAULT_LIMIT && ! $hasLimitParam) {
+                $paginationControl->setDefaultPageSize($paginationControl->getDefaultPageSize() * 2);
+
+                $paginationControl->apply();
+            }
```

Comment:
If the paginator uses a limit of `50`, the limiter does the same. So one of these checks should suffice. I'd say the paginator check should be kept, as it's the pagination what this is all about. So please merge the `if` above with the one below but drop its condition.